atomic_reactor/plugins/pre_inject_yum_repos.py

```diff
@@ -0,0 +1,264 @@
+        """
+        super().__init__(workflow)
+        self.target = target
+
+        self.xmlrpc = None
+        self.pathinfo = None
+        self.proxy = None
+        self.repourls = repourls or []
+        self.inject_proxy = inject_proxy
+        self.yum_repos = []
+        self.allowed_domains = self.workflow.conf.yum_repo_allowed_domains
+        self.include_koji_repo = False
+        self._builder_ca_bundle = None
+        self._ca_bundle_pem = None
+        self._dockerfile = None
+
+    def _inject_into_repo_files(self, build_dir: BuildDir):
+        """Inject repo files into a relative directory inside the build context"""
+        if build_dir.dockerfile.baseimage is None:
+            raise RuntimeError("No FROM line in Dockerfile")
+        host_repos_path = build_dir.path / RELATIVE_REPOS_PATH
+        self.log.info("creating directory for yum repos: %s", host_repos_path)
+        os.mkdir(host_repos_path)
+        allow_repo_dir_in_dockerignore(build_dir.path)
+
+        for repo in self.yum_repos:
+            # Update every repo accordingly in a repofile
+            # input_buf ---- updated ----> updated_buf
+            with StringIO(repo.content.decode()) as input_buf, StringIO() as updated_buf:
+                for line in input_buf:
+                    updated_buf.write(line)
+                    # Apply sslcacert to every repo in a repofile
+                    if line.lstrip().startswith('[') and self._builder_ca_bundle:
+                        updated_buf.write(f'sslcacert=/tmp/{self._ca_bundle_pem}\n')
+
+                yum_repo = YumRepo(repourl=repo.dst_filename,
+                                   content=updated_buf.getvalue(),
+                                   dst_repos_dir=host_repos_path,
+                                   add_hash=False)
+                yum_repo.write_content()
+
+    def _inject_into_dockerfile(self, build_dir: BuildDir):
+        self._dockerfile = build_dir.dockerfile
+        build_dir.dockerfile.add_lines(
+            "ADD %s* %s" % (RELATIVE_REPOS_PATH, YUM_REPOS_DIR),
+            all_stages=True, at_start=True, skip_scratch=True
+        )
+
+        if self._builder_ca_bundle:
+            shutil.copyfile(
+                self._builder_ca_bundle,
+                build_dir.path / self._ca_bundle_pem
+            )
+            build_dir.dockerfile.add_lines(
+                f'ADD {self._ca_bundle_pem} /tmp/{self._ca_bundle_pem}',
+                all_stages=True, at_start=True, skip_scratch=True
+            )
+
+        if not self.workflow.dockerfile_images.base_from_scratch:
+            build_dir.dockerfile.add_lines(*self._cleanup_lines())
+
+    def run(self):
+        """
+        run the plugin
+        """
+        if (self.workflow.dockerfile_images.base_from_scratch and
+                not self.workflow.dockerfile_images):
+            self.log.info("Skipping plugin, from scratch single stage can't add repos")
```

Comment:
this isn't just scratch single stage

there can be even multiple stages, but all scratch

Reply:
switched to check for `not self.workflow.dockerfile_images`